commit 22b6296be0846f77ef8b7bbce2e658c0f361da45
Author: Adrian Prantl <adrian@llnl.gov>
Date:   Tue Nov 6 15:56:23 2012 -0800

    remove field access from ref_borrow
    
    Conflicts:
    
    	compiler/AST/expr.cpp

diff --git a/runtime/include/chplrt.h b/runtime/include/chplrt.h
index 60a3e74..f832692 100644
--- a/runtime/include/chplrt.h
+++ b/runtime/include/chplrt.h
@@ -32,11 +32,17 @@ extern int chpl_threads_initialized;
 #define CHPL_ASSIGN_SVEC(x, y)                     \
   memcpy(&x, &y, sizeof(x))
 
+#define _CHECK_NIL(x, lineno, filename)                                 \
+  do {                                                                  \
+    if (x == nil)                                                       \
+      chpl_error("attempt to dereference nil", lineno, filename);	\
+  } while (0)
+
 #define _REF_BORROW(x, type, fieldname,data, lineno, filename)		\
   do {									\
     if (data == nil)							\
       chpl_error("data vector is nil", lineno, filename);		\
-  (x)->fieldname = (type*) data;					\
+    x = (type*) data;							\
   } while (0)
 
 #define _ARRAY_GET(x, i) (&((x)[i]))
